libraries/AP_HAL_ChibiOS/hwdef/common/malloc.c

```diff
@@ -635,7 +635,7 @@ bool mem_is_dma_safe(const void *addr, uint32_t size, bool filesystem_op)
         // use bouncebuffer for all non FS ops on H7
         return false;
     }
-    if (((uint32_t)addr) & 0x1F) {
+    if ((((uint32_t)addr) & 0x1F) != 0 || (size & 0x1F) != 0) {
```

Comment:
If you are going to check size for alignment, shouldn't you also do that for F4 et al below?

Reply:
it is only H7 that needs it I think. the reason we need to check size is so the flush/invalidate doesn't impact other data. We're only doing that for H7